rs/registry/canister/src/mutations/do_remove_node_operators.rs

```diff
@@ -73,3 +67,51 @@ impl Registry {
         }
     }
 }
+
+/// The payload of a request to remove Node Operator records from the Registry
+#[derive(Clone, Debug, Eq, PartialEq, CandidType, Deserialize, Serialize, Hash)]
+pub struct RemoveNodeOperatorsPayload {
+    // Old compatibility field, required for Candid, to be removed in the future
+    pub node_operators_to_remove: Vec<Vec<u8>>,
+
+    // New field, where the Node Operator IDs are passed as PrincipalIds instead of Vec<u8>
+    pub node_operator_principals_to_remove: Option<NodeOperatorPrincipals>,
+}
+
+/// Wrapper message for the optional repeated field
+#[derive(Clone, Debug, Eq, PartialEq, CandidType, Deserialize, Serialize, Hash)]
+pub struct NodeOperatorPrincipals {
+    pub principals: Vec<PrincipalId>,
+}
+
+impl RemoveNodeOperatorsPayload {
+    pub fn new(node_operators_to_remove: Vec<PrincipalId>) -> Self {
+        Self {
+            node_operators_to_remove: vec![],
+            node_operator_principals_to_remove: Some(NodeOperatorPrincipals {
+                principals: node_operators_to_remove,
+            }),
+        }
+    }
+
+    pub fn principal_ids_to_remove(&self) -> Vec<PrincipalId> {
+        // Request from Max:
```

Comment:
This doesn't seem like it should stay in the code :)

Reply:
Haha... alright, let me remove it
```suggestion
```